Marlin/Marlin_main.cpp

```diff
@@ -5414,6 +5388,188 @@ inline void gcode_M226() {
   }
 #endif
 
+/*
+* M242 - General I2C Message Interface
+*   A - 4 - 127 7-bit decimal device address
+*   P - 0 - 255 Process ID (See I2C Commands in Configuration_adv.h)
+*   D - 0 - 255 Data to write
+*   E - 0 - 255 EEPROM address, if applicable
+*/
+inline void gcode_M242() {
+  int verbose_level = code_seen('V') || code_seen('v') ? code_value_short() : 0;
+  if (verbose_level < 0 || verbose_level > 4) {
+    SERIAL_ECHOLNPGM("?(V)erbose Level is implausible (0-4).");
+    return;
+  }
+
+  // Used to see if we've been given arguments, and to warn you through the
+  // serial port if they're not seen.
+  bool hasA, hasP, hasD, hasE;
+  uint8_t i2c_address;
+  uint8_t i2c_process_id;
+  uint8_t i2c_eeprom_address;
+  uint8_t i2c_data;
+  
+  // Desired address for peripheral device
+  if (hasA = code_seen('A')) {
+    i2c_address = (uint8_t)code_value();
+  }
+  // Desired process command given
+  if (hasP = code_seen('P')) {
+    i2c_process_id = (uint8_t)code_value();
+  }        
+  // Desired data given
+  if (hasD = code_seen('D')) {
+    i2c_data = (uint8_t)code_value();
+  }
+
+  // EEPROM address, if applicable
+  if (hasE = code_seen('E')) {
+    i2c_eeprom_address = (uint8_t)code_value();
+  }
+
+  if (!hasA){
+    SERIAL_ECHOLNPGM("No peripheral address given");
+  }
+  if (!hasP){
+    SERIAL_ECHOLNPGM("No process command given");
+  }
+  if (!hasD){
+    SERIAL_ECHOLNPGM("No data given");
+  }
+  if (!hasE){
+    SERIAL_ECHOLNPGM("No eeprom address given");
+  }
+
+  I2C__GeneralCommand(i2c_address, i2c_process_id, i2c_eeprom_address, i2c_data);
```

Comment:
Looks like if any of the above parameters are not given (ie no address given), it remains uninitialized. The function `I2C__GeneralCommand()` is then called with faulty arguments. Perhaps at the end of these `bool` checks, if the parameters have not been given and they are essential, return